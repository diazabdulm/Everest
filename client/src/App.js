import React, { useEffect } from "react";
import { BrowserRouter, Redirect, Route, Switch } from "react-router-dom";
import { useDispatch } from "react-redux";

import { checkUserSession } from "./redux/userSlice";

import PrivateRoute from "./components/private-route/private-route.component";

import TasksPage from "./pages/tasks/tasks.component";
import LoginAndRegisterPage from "./pages/login-and-register/login-and-register.component";

export default function App() {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(checkUserSession());
  }, [dispatch]);

  return (
    <BrowserRouter>
      <Switch>
        <Route exact path="/login" component={LoginAndRegisterPage} />
        <PrivateRoute exact path="/:projectId" component={TasksPage} />
        <Redirect to="/all" />
      </Switch>
    </BrowserRouter>
  );
}
