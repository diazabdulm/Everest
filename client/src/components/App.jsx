import React, { useEffect } from "react";
import { Redirect, Route, Switch } from "react-router-dom";
import { useDispatch } from "react-redux";

import { checkUserSession } from "../redux/userSlice";

import PrivateRoute from "./PrivateRoute";

import TasksPage from "../pages/Tasks";
import LoginAndRegisterPage from "../pages/LoginAndRegister";

export default function App() {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(checkUserSession());
  }, [dispatch]);

  return (
    <Switch>
      <Route exact path="/login" component={LoginAndRegisterPage} />
      <PrivateRoute exact path="/:projectId" component={TasksPage} />
      <Redirect to="/all" />
    </Switch>
  );
}
