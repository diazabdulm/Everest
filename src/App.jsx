import React, { useEffect } from "react";
import { makeStyles } from "@material-ui/core";
import { useDispatch } from "react-redux";

import PublicRoute from "./components/PublicRoute";
import PrivateRoute from "./components/PrivateRoute";

import { checkUserSession } from "./redux/userSlice";

import SignInAndSignUpPage from "./views/SignInAndSignUpPage";
import TasksPage from "./views/Tasks";

const useStyles = makeStyles(theme => ({
  root: {
	display: "flex",
  }
}));

const App = () => {
  const classes = useStyles();
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(checkUserSession());
  }, [dispatch]);

  return (
    <div className={classes.root}>
      <PublicRoute exact path="/" component={SignInAndSignUpPage} />
      {/* <PrivateRoute path="/projects/:projectId" component={TasksPage} /> */}
    </div>
  );
};

export default App;
