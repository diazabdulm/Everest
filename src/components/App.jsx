import React from "react";
import { Route } from "react-router-dom";
import { makeStyles } from "@material-ui/core/styles";

import Header from "./Header";
import Sidebar from "./Sidebar";
import Content from "./Content";

const useStyles = makeStyles({
  root: {
    display: "flex"
  }
});

const App = () => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <Header />
      <Sidebar />
      <Route path={`/projects/:projectName`} component={Content} />
    </div>
  );
};

export default App;
