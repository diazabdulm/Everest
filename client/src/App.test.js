import React from "react";
import configureStore from "redux-mock-store";
import { shallow, mount } from "enzyme";
import { Provider, useDispatch } from "react-redux";
import { getDefaultMiddleware } from "@reduxjs/toolkit";

import App from "./App";

describe("App", () => {
  let wrapper;
  const mockStore = configureStore([...getDefaultMiddleware()]);

  beforeEach(() => {
    const store = mockStore({
      user: {},
      tasks: [],
      projects: [],
      visibilityFilter: ""
    });

    wrapper = mount(
      <Provider store={store}>
        <App />
      </Provider>
    );
  });

  it("render with redux state", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
