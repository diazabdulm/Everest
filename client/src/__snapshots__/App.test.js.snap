// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render with redux state 1`] = `ReactWrapper {}`;
